.circleci/config.yml

```diff
@@ -39,107 +58,150 @@ jobs:
     <<: *machine_defaults
     steps:
       - checkout
+      - download_task_test_data
+      - install_python3_caper_gcs
       - run: *make_tag
       - run:
           no_output_timeout: 300m
           command: |
             source ${BASH_ENV}
-
             cd dev/test/test_task/
-            rm -rf atac-seq-pipeline-test-data
-            export BOTO_CONFIG=/dev/null
-            gsutil -m cp -r gs://encode-pipeline-test-samples/encode-atac-seq-pipeline/atac-seq-pipeline-test-data .
-            ./download_hg38_fasta_for_test_ataqc.sh
+
             for wdl in test_*.wdl
             do
-              json=${wdl%.*}.json
-              result=${wdl%.*}.result.json
-              ./test.sh ${wdl} ${json} ${TAG}
-              if [[ $(wc -l "${result}" | awk '{print $1}') > 1 ]]; then
-                python -c "import sys; import json; data=json.loads(sys.stdin.read()); sys.exit(int(not data[u'match_overall']))" < ${result}
+              prefix=${wdl%.*}
+              input=${prefix}.json
+              metadata=${prefix}.metadata.json
+              caper run ${wdl} -i ${input} -m ${metadata} --docker ${TAG}
+              if [[ "${wdl}" != "test_choose_ctl.wdl" ]]; then
+                echo "Validating outputs of ${prefix}"
+                res=$(jq '.outputs["'${prefix}'.compare_md5sum.match_overall"]' "${metadata}")
+                [[ "$res" != true ]] && exit 100
               fi
-              rm -f ${result}
+              rm -f ${metadata}
             done
-            
+
   test_workflow_se:
     <<: *machine_defaults
     steps:
       - checkout
+      - install_python3_caper_gcs
       - run: *make_tag
       - run:
           no_output_timeout: 300m
           command: |
-            source ${BASH_ENV}
-            gcloud --quiet config set project ${GOOGLE_PROJECT_ID}
+            INPUT=ENCSR889WQX_subsampled.json
+
             cd dev/test/test_workflow/
-            echo ${GCLOUD_SERVICE_ACCOUNT_SECRET_JSON} > tmp_key.json
-            ./test_atac.sh ENCSR889WQX_subsampled.json tmp_key.json ${TAG}
-            python -c "import sys; import json; data=json.loads(sys.stdin.read()); sys.exit(int(not data[u'outputs'][u'atac.qc_json_ref_match']))" < ENCSR889WQX_subsampled.metadata.json
+            source ${BASH_ENV}
+
+            echo ${GCLOUD_SERVICE_ACCOUNT_SECRET_JSON} > tmp_secret_key.json
+            export GOOGLE_APPLICATION_CREDENTIALS=$PWD/tmp_secret_key.json
+            caper run ../../../atac.wdl \
+              --backend gcp --gcp-prj ${GOOGLE_PROJECT_ID} \
+              --backend-file backend_gcp_service_account.conf \
+              --out-gcs-bucket ${CAPER_OUT_DIR} --tmp-gcs-bucket ${CAPER_TMP_DIR} \
+              -i ${INPUT} -m metadata.json --docker ${TAG}
+
+            res=$(jq '.outputs["atac.qc_json_ref_match"]' metadata.json)
+            [[ "$res" != true ]] && exit 100
+            rm -f metadata.json
```

Comment:
I think this could be wrapped in a command, with the input file as a parameter?

Reply:
Thanks, that's a good point. Wrapped it in a comment.